src/main/java/com/eerussianguy/firmalife/blocks/BlockCheesewheel.java

```diff
@@ -0,0 +1,224 @@
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.IBlockAccess;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+import net.minecraftforge.items.ItemHandlerHelper;
+
+import com.eerussianguy.firmalife.ConfigFL;
+import com.eerussianguy.firmalife.init.AgingFL;
+import com.eerussianguy.firmalife.init.StatePropertiesFL;
+import mcp.MethodsReturnNonnullByDefault;
+import net.dries007.tfc.api.capability.food.CapabilityFood;
+import net.dries007.tfc.api.capability.size.IItemSize;
+import net.dries007.tfc.api.capability.size.Size;
+import net.dries007.tfc.api.capability.size.Weight;
+import net.dries007.tfc.objects.te.TETickCounter;
+import net.dries007.tfc.util.Helpers;
+import net.dries007.tfc.util.OreDictionaryHelper;
+
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
+public class BlockCheesewheel extends Block implements IItemSize
+{
+    public static final PropertyInteger WEDGES = StatePropertiesFL.WEDGES;
+    public static final PropertyEnum<AgingFL> AGE = StatePropertiesFL.AGE;
+    protected static final AxisAlignedBB CHEESEWHEEL_AABB = new AxisAlignedBB(0.0625D, 0.0D, 0.0625D, 0.9375D, 0.5D, 0.9375D); // This could have a more complex bounding box
+
+    private final Supplier<? extends Item> item;
+
+    public BlockCheesewheel(Supplier<? extends Item> item)
+    {
+        super(Material.CAKE);
+        this.setDefaultState(this.blockState.getBaseState().withProperty(WEDGES, 0).withProperty(AGE, AgingFL.FRESH));
+        this.setTickRandomly(true);
+        this.item = item;
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public AxisAlignedBB getBoundingBox(IBlockState state, IBlockAccess source, BlockPos pos)
+    {
+        return CHEESEWHEEL_AABB;
+    }
+
+    @Override
+    public boolean onBlockActivated(World worldIn, BlockPos pos, IBlockState state, EntityPlayer playerIn, EnumHand handIn, EnumFacing facing, float hitX, float hitY, float hitZ)
+    {
+        ItemStack stack = playerIn.getHeldItem(handIn);
+        if (OreDictionaryHelper.doesStackMatchOre(stack, "knife"))
+        {
+            if (!worldIn.isRemote)
+            {
+                stack.damageItem(1, playerIn);
+                return this.cutCheese(worldIn, pos, state, playerIn);
+            }
+        }
+        return true;
+    }
+
+    private boolean cutCheese(World worldIn, BlockPos pos, IBlockState state, EntityPlayer playerIn)
+    {
+        int i = state.getValue(WEDGES);
+        if (i < 3)
+            worldIn.setBlockState(pos, state.withProperty(WEDGES, i + 1), 3);
+        else
+            worldIn.setBlockToAir(pos);
+
+        ItemStack output = new ItemStack(item.get());
+        CapabilityFood.applyTrait(output, state.getValue(AGE).getTrait());
```

Comment:
This seems fine. I am nervous about getting values off the state this deep into the function (don't want nulls) but I don't see any reason why this would be null. Plus CapabilityFood null checks